agent/worker.go

```diff
@@ -568,9 +573,9 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		case v := <-ch:
 			task := v.(*api.Task)
 			if match(task) {
-				w.mu.Lock()
+				w.mu.RLock()
 				go w.taskManagers[task.ID].Logs(ctx, *subscription.Options, publisher)
```

Comment:
The task can be deleted from `taskManagers` in `reconcileTaskState`. Don't we need to check that the task still exists in the map before invoking the `Logs` method?